test/system/fidelity_test_03.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+RED='\e[41m'
+GREEN='\e[42m'
+NC='\033[0m' # No Color
+BOLD='\033[1m'
+UNDERLINE='\033[4m'
+
+print_header() {
+    echo
+    echo "**************************************************"
+    echo "* Starting Fidelity Test 03 with path $directory_path"
+    echo "**************************************************"
+    echo "Test 03: Checks if the events have been written in sequential order."
+    echo
+}
+
+# Check if the correct number of arguments are provided
+if [ "$#" -ne 2 ]; then
+    echo "Usage: $0 <input_file> <directory>"
+    exit 1
+fi
+
+verify_order() {
+    local file1="$1"
+    local file2="$2"
+
+    # Check if both files are specified
+    if [[ -z "$file1" || -z "$file2" ]]; then
+        echo "Error: Two file paths must be provided"
+        return 1
+    fi
+
+    # Read file1 line by line
+    local current_line_number=0
+    local found
+    local last_line_number=0
+
+
+    while IFS= read -r line_from_file1; do
+        found=0
+        # Increment to move to the next line to be checked after a successful match
+        ((current_line_number++))
+        local line_num=$(echo "$line_from_file1" | cut -d ':' -f6)
+
+        # Search for the line in file2 starting from the last matched line
+        while IFS= read -r line_from_file2; do
+            if [[ "$line_num" == "$line_from_file2" ]]; then
+                found=1
+                # Update last_line_number to the line number of the matched line in file2
+                last_line_number="$current_line_number"
+                break
+            fi
+            ((current_line_number++))
+        done < <(tail -n +"$current_line_number" "$file2")
+
+        # If no match is found, the test fails
+        if [[ $found -eq 0 ]]; then
+            echo -e "\t${BOLD}Result:${NC} ${RED}Failed.${NC}\n"
```

Comment:
since we do ordering check file by file only to enable pin-pointing failing keepers, it would be better to report the failing keeper host name here.

Reply:
My understanding about this comment is that we want the test to report the file that is failing. But this is something that is already done. The test goes file by file and provides the following info:
Filename (This includes the keeper host name)
Date (Sometimes we have older files that may confuse the result)
Result: Success/Failed